cms/models.py

```diff
@@ -255,6 +266,76 @@ class FacultyMembersPage(CourseProgramChildPage):
     ]
 
 
+class HomePage(Page):
+    """
+    CMS Page representing the home/root route
+    """
+
+    # disable promote panels, no need for slug entry, it will be autogenerated
+    promote_panels = []
+
+    subpage_types = [
+        "CoursePage",
+        "ProgramPage",
+        "CoursesInProgramPage",
+        "LearningTechniquesPage",
+        "UserTestimonialsPage",
+        "ForTeamsPage",
+    ]
+
+    def _get_child_page_of_type(self, cls):
+        """Gets the first child page of the given type if it exists"""
+        child = self.get_children().type(cls).first()
+        return child.specific if child else None
+
+    @property
+    def learning_experience(self):
+        """
+        Gets the "Learning Experience" section subpage
+        """
+        return self._get_child_page_of_type(LearningTechniquesPage)
+
+    @property
+    def testimonials(self):
+        """
+        Gets the testimonials section subpage
+        """
+        return self._get_child_page_of_type(UserTestimonialsPage)
+
+    @property
+    def upcoming_courseware(self):
+        """
+        Gets the upcoming courseware section subpage
+        """
+        return self._get_child_page_of_type(CoursesInProgramPage)
+
+    @property
+    def inquiry_section(self):
+        """
+        Gets the "inquire now" section subpage
+        """
+        return self._get_child_page_of_type(ForTeamsPage)
+
+    def save(self, *args, **kwargs):
+        # autogenerate a unique slug so we don't hit a ValidationError
+        self.slug = slugify("home-{}".format(self.title))
+        super().save(*args, **kwargs)
+
+    def get_context(self, request, *args, **kwargs):
+        from courses.models import Program
```

Comment:
Is that import above causing any issue ?

Reply:
causes a circular import if I put it at the top of the module